### PR TITLE
Updated the branding and execution of Iris docs

### PR DESCRIPTION
**Pull request by pelson**
This includes:

 * removing the "easy-to-use" statement in Iris' strapline. Let's let users be the judge of that.
 * improved the styling of the homepage, including re-instating the search bar. (I removed it, and consider it to be a mistake until we properly integrate it into the navigation panel)
 * Added a what's new for #3010.
 * Added a extlinks section for convenient linking to github issues with ```:issue:`number` ``` and ```:pr:`number` ```.

### CODE REVIEW

**Comment by pelson** [comment]
@bjlittle - FWY, this is the last PR for v2.1.0. #3038.

File: docs/iris/src/_templates/index.html [hunk 1]
```diff
@@ -13,15 +13,6 @@
 #sliderNav div { display: inline-block; height: 8px; width: 8px; margin: 0 2px;
   border: #ccc medium solid; border-radius: 50%; box-shadow: 0 0 2px 0 #ccc;}
 #sliderNav div.activeSlide { border-color: #777; background-color: #ccc; }
-
-
-.bodywrapper {
```
Comment (pelson) [inline docs/iris/src/_templates/index.html]: This was removed, as I added it to hide the search bar. I think that is premature for the release.
Long-term, I'd like to move the search to the nav bar at the top.

File: docs/iris/src/_templates/layout.html [hunk 2]
```diff
@@ -23,49 +23,24 @@
 
 {% block relbar1 %}
 
-
-
-
-<style>
-.github-img {
```
Comment (pelson) [inline docs/iris/src/_templates/layout.html]: These were moved into style.css.

**Comment by pelson** [comment]
I've pushed a demo of the docs at https://rawgit.com/pelson/scitools.org.uk/iris_docs_dev_tweak/iris/docs/phil_dev/index.html. These will pretty much be what is pushed for the v2.1 release, so open to any/all corrections at this point. I strongly recommend this being done in the form of a PR against my PR, rather than me having to interpret exactly what is intended from review comments.

cc @dkillick

**Comment by dkillick** [comment]
> I've pushed a demo of the docs

Thanks @pelson!

File: docs/iris/src/whatsnew/2.1.rst [hunk 3]
```diff
@@ -53,23 +53,24 @@ Iris 2.1 Features
...
-  the ``standard_parallel`` keyword argument.
+  the ``standard_parallel`` keyword argument. (:pr:`3041`)
 
 Bugs Fixed
 ==========
 
 * All var names being written to NetCDF are now CF compliant.
   Non alpha-numeric characters are replaced with '_', and var names now always
-  have a leading letter. https://github.com/SciTools/iris/pull/2930
+  have a leading letter. (:pull:`2930`)
```
Comment (dkillick) [inline docs/iris/src/whatsnew/2.1.rst]: It's a stylistic thing only but I'd prefer to see the brackets containing the link _inside_ the sentence; something like:

```
* :mod:`iris.quickplot` labels now honour the axes being drawn to when using the ``axes`` keyword (:pull:`3010`).
```

(I've chosen a shorter example from below for the sake of brevity.)

File: docs/iris/src/_templates/layout.html [hunk 4]
```diff
@@ -23,49 +23,24 @@
...
-      <h1 style="font-size: 3.5rem; font-family: 'Alike', serif; margin-top: 40px; padding: 0px; color: #323232; padding-bottom: 0.2em;">
-          Iris <span style="font-size: 1.5rem;">v2.1</span>
+    <div class="strapline">
+      <h1>
+          Iris <span class="version">v2.1</span>
       </h1>
-      <h4 style="padding: 0px; max-width: 60ch; margin-top: 0px; margin-bottom: 0px;">
-         a powerful, easy to use, community-driven Python library for analysing and
+      <h2>
+         a powerful, format-agnostic, and community-driven Python library for analysing and
```
[the remark this answers is not in the document]
Comment (kaedonkers) [reply docs/iris/src/_templates/layout.html]: I second this!

File: README.md [hunk 5]
```diff
@@ -5,7 +5,7 @@
 </h1>
 
 <h4 align="center">
-    Iris is a powerful, easy to use, and community-driven Python library for
+    Iris is a powerful, format-agnostic, and community-driven Python library for
```
[the remark this answers is not in the document]
Comment (pelson) [reply README.md]: Yep. Will do now that it is reviewed 👍

File: docs/iris/src/_static/style.css [hunk 6]
```diff
@@ -1,3 +1,7 @@
+body {
+    font-family: 'Noto Sans', sans-serif;
```
Comment (pelson) [inline docs/iris/src/_static/style.css]: Nice choice @dkillick.

**Comment by dkillick** [comment]
@pelson the docs look a lot nicer now 😁

**Pull request merged**